src/Abc.Zebus/Core/MessageContextAwareBus.cs

```diff
@@ -91,6 +91,14 @@ public void Reply(IMessage response)
             remove => _bus.Stopped -= value;
         }
 
+        public void Publish(IEvent message, PeerId targetPeer)
+        {
+            if (_bus is IInternalBus bus)
+                bus.Publish(message, targetPeer);
+            else
+                throw new NotImplementedException("Inner bus is not IInternalBus");
```

Comment:
We should never provide a non-internal bus to the `MessageAwareContextBus` thus it's an error message designed for someone developing in the Zebus codebase.

Reply:
It could still happen if the user overrides the bus in the container, but I agree it's not very likely.

The comment on the exception type is still valid.